go_modules/lib/dependabot/go_modules/file_parser.rb

```diff
@@ -50,6 +62,41 @@ def ecosystem
 
       private
 
+      sig { void }
+      def set_go_environment_variables
+        set_goenv_variable
+        set_goprivate_variable
+        set_goproxy_variable
+      end
+
+      sig { void }
+      def set_goenv_variable
+        return unless go_env
+
+        env_file = T.must(go_env)
+        File.write(env_file.name, env_file.content)
+        ENV["GOENV"] = Pathname.new(env_file.name).realpath.to_s
+      end
+
+      sig { void }
+      def set_goprivate_variable
+        return if go_env&.content&.include?("GOPRIVATE")
+
+        goprivate = options.fetch(:goprivate, "*")
+        ENV["GOPRIVATE"] = goprivate if goprivate
+      end
+
+      sig { void }
+      def set_goproxy_variable
+        return if go_env&.content&.include?("GOPROXY")
+
+        goproxy_credentials = credentials.select { |cred| cred["type"] == "goproxy_server" }
```

Comment:
where do these `goproxy_server` credentials come from? In the go tests I see `git_source` credentials being used.